File: patterns/C++/pattern7.cpp
/*
Generate the following patterns
      *           |      *
    * * *         |     *** 
  * * * * *       |    ***** 
* * * * * * *     |   ******* 
*/

#include<bits/stdc++.h>
using namespace std;

void pattern7_1(int n){
    for(int i=0; i<n; i++){
        for(int j=0; j<2*(n-i-1); j++){
            cout << " ";
        }

        for(int k=0; k<2*i+1; k++){
            cout << "* ";
        
        }
        cout << "\n";
    }
}

void pattern7_2(int n){
    for(int i=0; i<n; i++){
        for(int j=0; j<(n-i-1); j++){
            cout << " ";
        }

        for(int k=0; k<2*i+1; k++){
            cout << "*";
        
        }
        cout << "\n";
    }
}

int main() {
    int n;
    cout << "Enter the number of rows(n): ";
    cin >> n;
    cout << "Here is the generated pattern\n";
    pattern7_1(n);
    pattern7_2(n);


}

File: patterns/Python/pattern7.py
"""
Generate the following pattern
      *           |      *
    * * *         |     *** 
  * * * * *       |    ***** 
* * * * * * *     |   ******* 
"""

def pattern7_1(n):
    for i in range(n):
        for j in range(2*(n-1-i)):
            print(" ", end="")
        for k in range(2*i+1):
            print("* ", end="")
        print("")
        
        
def pattern7_2(n):
    for i in range(n):
        for j in range(n-i-1):
            print(" ", end="")
        for k in range(2*i+1):
            print("*", end="")
        print("")
        

def main():
    i = int(input("Enter the number of rows: "))
    print("Here is the generated pattern\n")
    pattern7_1(i)
    print("")
    pattern7_2(i)
    
if __name__ == "__main__":
    main()